src/aposteriori/dnn/data_processing/tools.py

```diff
@@ -1,14 +1,68 @@
 
 from ampal.amino_acids import standard_amino_acids
 from collections import Counter
 from pathlib import Path
 from random import shuffle
 
-from aposteriori.dnn.config import UNCOMMON_RESIDUE_DICT
+from aposteriori.data_prep.create_frame_data_set import DatasetMetadata
+from aposteriori.dnn.config import UNCOMMON_RESIDUE_DICT, MAKE_FRAME_DATASET_VER
+
+
+def extract_metadata_from_dataset(frame_dataset: Path) -> DatasetMetadata:
+    """
+    Retrieves the metadata of the dataset and does a sanity check of the version.
+    If the dataset version is not compatible with aposteriori, the training process will stop.
+
+    Parameters
+    ----------
+    frame_dataset: Path
+        Path to the .h5 dataset with the following structure.
+        └─[pdb_code] Contains a number of subgroups, one for each chain.
+          └─[chain_id] Contains a number of subgroups, one for each residue.
+            └─[residue_id] voxels_per_side^3 array of ints, representing element number.
+              └─.attrs['label'] Three-letter code for the residue.
+              └─.attrs['encoded_residue'] One-hot encoding of the residue.
+        └─.attrs['make_frame_dataset_ver']: str - Version used to produce the dataset.
+        └─.attrs['frame_dims']: t.Tuple[int, int, int, int] - Dimentsions of the frame.
+        └─.attrs['atom_encoder']: t.List[str] - Lables used for the encoding (eg, ["C", "N", "O"]).
+        └─.attrs['encode_cb']: bool - Whether a Cb atom was added at the avg position of (-0.741287356, -0.53937931, -1.224287356).
+        └─.attrs['atom_filter_fn']: str - Function used to filter the atoms in the frame.
+        └─.attrs['residue_encoder']: t.List[str] - Ordered list of residues corresponding to the encoding used.
+        └─.attrs['frame_edge_length']: float - Length of the frame in Angstroms (A)
+
+    Returns
+    -------
+    dataset_metadata: DatasetMetadata of the dataset with the following parameters:
+        make_frame_dataset_ver: str
+        frame_dims: t.Tuple[int, int, int, int]
+        atom_encoder: t.List[str]
+        encode_cb: bool
+        atom_filter_fn: str
+        residue_encoder: t.List[str]
+        frame_edge_length: float
+
+    """
+    with h5py.File(frame_dataset, "r") as dataset_file:
+        meta_dict = dict(dataset_file.attrs.items())
+        dataset_metadata = DatasetMetadata.import_metadata_dict(meta_dict)
+
+    # Extract version metadata:
+    dataset_ver_num = dataset_metadata.make_frame_dataset_ver.strip(".")[0]
+    aposteriori_ver_num = MAKE_FRAME_DATASET_VER.strip(".")[0]
+    # If the versions are compatible, return metadata
+    if dataset_ver_num == aposteriori_ver_num:
+        return dataset_metadata
+    else:
+        sys.exit(
+            f"Dataset version is {dataset_metadata.make_frame_dataset_ver} and is incompatible "
+            f"with Aposteriori version {MAKE_FRAME_DATASET_VER}."
+            f"Try re-creating the dataset with the current version of Aposteriori."
+        )
```

Comment:
If possible, I think that it's best to have the intended execution path to have a return statement at the end of the function body. I think that it's easier to read and reason about.

```suggestion
    if dataset_ver_num != aposteriori_ver_num:
        sys.exit(
            f"Dataset version is {dataset_metadata.make_frame_dataset_ver} and is incompatible "
            f"with Aposteriori version {MAKE_FRAME_DATASET_VER}."
            f"Try re-creating the dataset with the current version of Aposteriori."
        )
    return dataset_metadata
```